utils/key/key.go

```diff
@@ -37,34 +61,49 @@ func (k basicKey) Append(suffix Key) Key {
 	}
 }
 
-func (k basicKey) Bytes(delimiter ...string) []byte {
-	del := DefaultDelimiter
-	if len(delimiter) == 1 {
-		del = delimiter[0]
-	}
-	return bytes.Join(k.particles, []byte(del))
+func (k basicKey) Bytes() []byte {
+	return bytes.Join(k.particles, []byte(DefaultDelimiter))
 }
 
 func (k basicKey) String() string {
 	return string(k.Bytes())
 }
 
 // FromBz creates a new Key from bytes
+// Deprecated: use FromBzHashed instead
 func FromBz(key []byte) Key {
 	return &basicKey{particles: [][]byte{key}}
 }
 
 // FromStr creates a new Key from a string
+// Deprecated: use FromStrHashed instead
 func FromStr(key string) Key {
 	return FromBz([]byte(strings.ToLower(key)))
 }
 
 // From creates a new Key from a fmt.Stringer interface
+// Deprecated: use FromHashed instead
 func From(key fmt.Stringer) Key {
 	return FromStr(key.String())
 }
 
 // FromUInt creates a new Key from any unsigned integer type
 func FromUInt[T constraints.Unsigned](key T) Key {
-	return FromBz(convert.IntToBytes(key))
+	return &basicKey{particles: [][]byte{convert.IntToBytes(key)}}
+}
+
+// FromBzHashed creates a new Key from bytes
+func FromBzHashed(key []byte) Key {
+	hash := sha3.Sum256(key)
+	return basicKey{particles: [][]byte{hash[:]}}
+}
+
+// FromStrHashed creates a new Key from a string
+func FromStrHashed(key string) Key {
+	return FromBzHashed([]byte(strings.ToLower(key)))
+}
+
+// FromHashed creates a new Key from a fmt.Stringer interface
+func FromHashed(key fmt.Stringer) Key {
```

Comment:
```suggestion
func FromStringerHashed(key fmt.Stringer) Key {
```

Reply:
I'm using the same naming scheme as before and the idea is to rename it to the old name once we migrated. I just didn't want to do a rename here because it would mean that 50ish files will change